Sources/AppStoreConnectCLI/Model/PrereleaseVersion.swift

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 Itty Bitty Apps Pty Ltd
+
+import AppStoreConnect_Swift_SDK
+import Foundation
+import SwiftyTextTable
+
+struct PreReleaseVersionDetails: ResultRenderable {
+    var app: App?
+    var platform: String?
+    var version: String?
+    // TODO: var builds: [Build]?
+}
+
+extension PreReleaseVersionDetails {
+    init(_ preReleaseVersion: AppStoreConnect_Swift_SDK.PrereleaseVersion, _ includes: [AppStoreConnect_Swift_SDK.PreReleaseVersionRelationship]?) {
+
+        let relationships = preReleaseVersion.relationships
+
+        let includedApps = includes?.compactMap { relationship -> AppStoreConnect_Swift_SDK.App? in
+          if case let .app(app) = relationship {
+            return app
+          }
+          return nil
+        }
+
+        let appDetails = includedApps?.filter { relationships?.app?.data?.id == $0.id }.first
```

Comment:
```suggestion
        let appDetails = includedApps?.first { relationships?.app?.data?.id == $0.id }
```

Reply:
I don't think this is correct. You should first filter by condition and then take the first occurrence.

Reply:
I think `array.filter` will iterate the array and `array.first` will find the first then stop looking the others. I guess `array.first` will have a better performance? @adamjcampbell What's your thought on this?

Reply:
I think filtering first is more safer because if you are only considering the first occurrence and the one that you want is not at the first occurrence then you are missing out the one you actually want.

Reply:
@Nafisa2020 this is actually the function [`first(where:)`](https://developer.apple.com/documentation/swift/array/1848165-first)
It will iterate and stop when it finds the first thing that matches, it's a very useful function

Reply:
then it makes sense otherwise blindly assuming it will be always at the first occurrence is not good.

Reply:
Yeh it's actually the func `first(where: )`, sometimes omit the keyword `where` is not a good idea 🤔